docs/INSTALL-devel.rst

```diff
@@ -0,0 +1,292 @@
+typical default postgres setup, the right value is ``localhost``).
+
+(Configuring other sections of **haas.cfg** is explained in section `Configuring HaaS`_.)  
+
+
+:Continue with installation steps in the next section.
+
+
+
+
+Getting Started with HAAS Installation
+======================================
+First you will need to fork and clone the HaaS repo into your dev VM.::
+
+  git clone https://github.com/**username**/haas.git
+  cd haas
+
+
+The first time you start working in the repository, set up a clean test
+environment::
+
+  virtualenv .venv
+
+Enter the environment (do this every time you start working with HaaS dev environment)::
+
+  source .venv/bin/activate
+
+Then, proceed with installing the HaaS and its dependencies into the virtual
+environment::
+
+  sudo python setup.py install
+  pip install -e .
+
+
+For older systems:
+==================
+
+On systems with older versions of ``pip``, such as Debian Wheezy and Ubuntu
+12.04, this install will fail with the following error::
+
+  AttributeError: 'NoneType' object has no attribute 'skip_requirements_regex'
+
+Fix this by upgrading ``pip`` within the virtual environment::
+
+  pip install --upgrade pip
+
+Versions of python prior to 2.7 don't have importlib as part of their
+standard library, but it is possible to install it separately. If you're
+using python 2.6 (which is what is available on CentOS 6, for example),
+you may need to run::
+
+  pip install importlib
+
+
+Setting up the Database:
+========================
+By default dev environment uses SQLite as a database, so if you're using it you can skip this section.
+
+If you wish to use postgreSQL instead, you may get an error ``psycopg2 package not found``.
+  
+When you do ``haas-admin db create`` in the next step, you may need to install 
+following package on your system 
+
+CentOS::  
+
+  yum install postgresql-devel
+
+Ubuntu::
+  
+  sudo apt-get install libpq-dev
+
+before installing ``psycopg2`` in the virtualenv for HaaS::
+
+  pip install psycopg2
+
+
+Configuring HaaS
+================
+
+Now the ``haas`` executable should be in your path.  First, create a
+configuration file ``haas.cfg``. There are two examples for you to work from,
+``examples/haas.cfg.dev-no-hardware``, which is oriented towards development, and
+``examples/haas.cfg`` which is more production oriented.  These config
+files are well commented; read them carefully.
+
+HaaS can be configured using ``haas.cfg`` to not perform state-changing operations on nodes,
+headnodes and networks, allowing developers to run and test parts of a haas
+server without requiring physical hardware. To suppress actual node and headnode
+operations, set ``dry_run = True`` in the ``[devel]`` section. 
+
+
+Most customization require including directives under section ``[extensions]``
+
+For suppressing actual network switch operations, use the ``mock`` switch driver :: 
+  haas.ext.switches.mock =
```

Comment:
I would be more comfortable if we do:
   "`` haas.ext.switches.mock`` is adopted to suppress actual network switch operations."

It would be more consistency if we only use ``::`` on commands. The same to the ``haas.ext.auth.null`` and ``haas.ext.auth.database``.